modules/picard_qc/1.0/picard_qc.smk

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Lauren Chong
+# Module Author:    Helena Winata
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["picard_qc"]`
+CFG = op.setup_module(
+    name = "picard_qc",
+    version = "1.0",
+    subdirectories = ["inputs", "metrics", "merged_metrics", "outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _picard_qc_input_bam,
+    _picard_qc_rrna_int,
+    _picard_qc_merge_metrics,
+    _picard_qc_merged_output,
+    _picard_qc_flagstats_output,
+    _picard_qc_merged_dispatch,
+    _picard_qc_all
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _picard_qc_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"]
```

Comment:
Are BAM indices required? If so, I would follow the same approach as `manta` 2.1. Otherwise, if the user is generating the BAM files themselves using custom rules and the BAM index in a separate rule, this won't trigger the latter.

Reply:
Hmm I think it uses the default (same directory are the original bam files) but I will ads .bai symlinking